### PR TITLE
Mat 5763 relative data types

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-5763](https://jira.cms.gov/browse/MAT-5763)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataTypes.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Button } from "@madie/madie-design-system/dist/react";
+import AddCircleOutlineIcon from "@mui/icons-material/AddCircleOutline";
+
+export default function DataTypes({ selectedDataElements }) {
+  const capitalizeFirstLetter = (w) => w.charAt(0).toUpperCase() + w.slice(1);
```
Comment (adongare) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataTypes.tsx]: looks like this is lodash `_.capitalize` https://lodash.com/docs/4.17.15#capitalize

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DynamicElementTabs.tsx [hunk 2]
```diff
@@ -1,47 +1,83 @@
...
-      scrollButtons={false}
-    >
-      {/* have to do a ts ignore line because it's receiving a forward ref from Tabs
+    <div>
+      <Tabs
+        id="categories-tabs"
+        value={activeTab}
+        onChange={(e, v) => {
+          const elements: DataElement[] = findDataElements(v);
+          setSelectedDataElements(elements);
```
Comment (adongare) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DynamicElementTabs.tsx]: I don't think this is necessary. useEffect hook would take care of this.

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DynamicElementTabs.tsx [hunk 3]
```diff
@@ -1,47 +1,83 @@
...
+    if (activeTab) {
+      setSelectedDataElements(findDataElements(activeTab));
+    }
+  }, [activeTab]);
+
+  const findDataElements = (category: string): DataElement[] => {
+    const elements: DataElement[] = dataElements?.filter(
+      (element) => element.qdmCategory === category
+    );
+    return elements;
```
Comment (adongare) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DynamicElementTabs.tsx]: this will display race, ethnicity, and gender as well. please check with Stan/Nicole if that is needed.

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataTypes.tsx (hunk 1 above)
Comment (sb-cecilialiu) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataTypes.tsx]: @adongare thanks will do.

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DynamicElementTabs.tsx (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DynamicElementTabs.tsx]: @adongare You are right the useEffect does take care of this, will remove, thanks!

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DynamicElementTabs.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DynamicElementTabs.tsx]: @adongare I just removed race, ethnicity and gender. Thanks Ashok!

**Review by adongare — APPROVED** [review]
looks good!
